core/shared/src/main/scala/io/circe/Error.scala

```diff
@@ -20,6 +22,11 @@ final object ParsingFailure {
   implicit final val eqParsingFailure: Eq[ParsingFailure] = Eq.instance {
     case (ParsingFailure(m1, t1), ParsingFailure(m2, t2)) => m1 == m2 && t1 == t2
   }
+
+  implicit final val showParsingFailure: Show[ParsingFailure] = Show.show { failure =>
+    val name = classOf[ParsingFailure].getSimpleName
```

Comment:
Yeah for staying safe if ever refactoring. But this is easily caught by tests instead, so let's go with hard coded names.